server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -156,9 +157,11 @@ static ClusterState createDataStream(MetadataCreateIndexService metadataCreateIn
         assert firstBackingIndex != null;
         assert firstBackingIndex.mapping() != null : "no mapping found for backing index [" + firstBackingIndexName + "]";
 
-        String fieldName = template.getDataStreamTemplate().getTimestampField();
+        String fieldName = composableTemplate.getDataStreamTemplate().getTimestampField();
         DataStream.TimestampField timestampField = new DataStream.TimestampField(fieldName);
-        DataStream newDataStream = new DataStream(request.name, timestampField, List.of(firstBackingIndex.getIndex()));
+        Template template = composableTemplate.template();
+        boolean hidden = template != null && template.settings() != null && IndexMetadata.INDEX_HIDDEN_SETTING.get(template.settings());
```

Comment:
I feel that reusing the `index.hidden` setting is adding ambiguity to the setting, since it will then be controlling whether an index is hidden or data stream depending on the context how it is being used.

I would prefer that a data stream can be made hidden via the `data_stream` section of a composable index template. I think this way it is clearer what is being made hidden. Currently `DataStreamTemplate` is empty, but I think we will be adding more configuration options that are data stream specific later (e.g. timestamp field name). 

Note that aliases are also made hidden via a property on an alias instead of reusing the `index.hidden` setting. So making data stream hidden via a parameter on `DataStreamTemplate` is inline how aliases
can be made hidden.

Let me know what you think. I think this change would be a minor change to this PR.

Reply:
That makes sense, I'll change it

Reply:
It's done, now it uses `data_stream` part of a template
```json
"data_stream": {
    "hidden": true
}
```